grpcclient/grpcclient.go

```diff
@@ -91,13 +98,29 @@ func (cfg *Config) DialOption(unaryClientInterceptors []grpc.UnaryClientIntercep
 
 	return append(
 		opts,
-		grpc.WithDefaultCallOptions(cfg.CallOptions()...),
-		grpc.WithUnaryInterceptor(middleware.ChainUnaryClient(unaryClientInterceptors...)),
-		grpc.WithStreamInterceptor(middleware.ChainStreamClient(streamClientInterceptors...)),
-		grpc.WithKeepaliveParams(keepalive.ClientParameters{
-			Time:                time.Second * 20,
-			Timeout:             time.Second * 10,
+		withDefaultCallOptions(cfg.CallOptions()...),
+		withUnaryInterceptor(middleware.ChainUnaryClient(unaryClientInterceptors...)),
+		withStreamInterceptor(middleware.ChainStreamClient(streamClientInterceptors...)),
+		withKeepaliveParams(keepalive.ClientParameters{
+			Time:                time.Duration(cfg.KeepaliveTime) * time.Second,
+			Timeout:             time.Duration(cfg.KeepaliveTimeout) * time.Second,
 			PermitWithoutStream: true,
 		}),
 	), nil
 }
+
+var withDefaultCallOptions = func(cos ...grpc.CallOption) grpc.DialOption {
```

Comment:
Instead of using global variables for these, can they be passed to `dialOption` as parameters? (Default publicly-exported `DialOption` would just use grpc-functions). This avoids needing setup and cleanup, and can be hidden into “fakeDialOption” method or similar.

Reply:
I think this feedback hasn't been addressed, am I wrong?

Reply:
@pracucci you're right! I was deferring addressing this until consensus on whether the PR should be closed or not.

Reply:
@pracucci @pstibrany the drawback I see with stubbing these dependencies via arguments is that it adds complexity. The tests will no longer call `Config.DialOption` itself, and introduces the risk of having a bug in `Config.DialOption` where it passes the wrong dependencies. The test coverage will also be lower with this technique.

If you still prefer the injection via method arguments technique, I can change to that. Just want to clarify the problems I see with it.